tests/smartswitch/common/device_utils_dpu.py

```diff
@@ -0,0 +1,184 @@
+"""
+Helper script for DPU  operations
+"""
+import logging
+import pytest
+from tests.common.devices.sonic import *  # noqa: F401,F403
+from tests.platform_tests.api.conftest import *  # noqa: F401,F403
+from tests.common.helpers.platform_api import chassis, module
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+from pkg_resources import parse_version
+
+
+@pytest.fixture(scope='function')
+def num_dpu_modules(platform_api_conn):
+
+    """
+    Returns the number of DPU modules
+    """
+
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+    return num_modules
+
+
+@pytest.fixture(scope='function')
+def skip_test_smartswitch(duthosts, enum_rand_one_per_hwsku_hostname,
+                          platform_api_conn):
+    """
+    Checks whether given testbed is running
+    202405 image or below versions
+    If True, then skip the script
+    else checks if dpus are in darkmode
+    If dpus are in dark mode, then power up the DPUs
+    else, proceeds to run all test cases
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    if not duthost.facts["DPUS"] and \
+            parse_version(duthost.os_version) <= parse_version("202405"):
+        pytest.skip("Test is not supported for this testbed and os version")
+
+    darkmode = is_dark_mode(duthost, platform_api_conn)
+
+    if darkmode:
+        dpu_power_on(duthost, platform_api_conn)
+
+
+def is_dark_mode(duthost, platform_api_conn):
```

Comment:
Function name is not clear. could you modify as is_dark_mode_enabled seems more related?

Reply:
Changed it.